kitsune/questions/views.py

```diff
@@ -617,9 +631,14 @@ def aaq_step2(request, product_key):
     return aaq(request, product_key=product_key, step=2)
 
 
-@login_required
 def aaq_step3(request, product_key, category_key=None):
     """Step 3: Show full question form."""
+
+    # Since removing the @login_required decorator for MA form
+    # need to catch unauthenticated, non-MA users here """
+    if product_key not in settings.LOGINLESS_PRODUCTS and not request.user.is_authenticated:
```

Comment:
Maybe instead of doing this check (which is a repeated block of code), you could supply an optional parameter in the function signature (eg `is_loginless=False` or `needs_login=True`) and modify that flag from the calling function. According to the flag state you may or may not redirect to a login wall.

Reply:
Got it.